kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection/subPages/PreviewSelectedResources.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+
+  <div>
+    <KCircularLoader v-if="loading" />
+    <PreviewContent
+      v-else
+      :currentContentNode="contentNode"
+      :ancestors="ancestors"
+      :isSelected="isSelected"
+      :questions="questions"
+      @addResource="handleAddResource"
+      @removeResource="handleRemoveResource"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  import { getCurrentInstance } from 'vue';
+  import commonCoreStrings from 'kolibri/uiText/commonCoreStrings';
+  import useFetchContentNode from '../../../../../../composables/useFetchContentNode';
+  import useResourceSelection from '../../../../../../composables/useResourceSelection';
+  import { coachStrings } from '../../../../../common/commonCoachStrings';
+  import { PageNames } from '../../../../../../constants/index';
+  import PreviewContent from '../../PreviewContent';
+
+  export default {
+    name: 'PreviewSelectedResources',
+    components: {
+      PreviewContent,
+    },
+    mixins: [commonCoreStrings],
+    setup(props) {
+      const { contentNode, ancestors, questions, loading } = useFetchContentNode(props.contentId);
+      const { selectedResources, selectResources, deselectResources } = useResourceSelection();
+      const { manageLessonResourcesTitle$ } = coachStrings;
+      const instance = getCurrentInstance();
+
+      props.setTitle(manageLessonResourcesTitle$());
+      props.setGoBack(() => {
+        return instance.proxy.$router.go(-1);
+      });
+
+      return {
+        contentNode,
+        ancestors,
+        questions,
+        selectedResources,
+        selectResources,
+        deselectResources,
+        loading,
+      };
+    },
+    props: {
+      contentId: {
+        type: String,
+        required: true,
+      },
+      setTitle: {
+        type: Function,
+        default: () => {},
+      },
+      setGoBack: {
+        type: Function,
+        default: () => {},
+      },
+    },
+    data() {
+      return {
+        justRemovedResource: false,
+      };
+    },
+    computed: {
+      isSelected() {
+        if (this.justRemovedResource) {
```

Comment:
I am not understanding well how does this work? What is the expected behaviour when we have a selected/deselected resource? Im not finding that in the specs cc: @marcellamaki

Reply:
Hmm I am not sure I am answering your question well, @AlexVelezLl but the preview is _mostly_ separate from the selected/de-selected aside from the "Add" button in the top right corner. So, previewing a resource on its own is not depending on the selection state. 

If the resource is _not_ currently in the lesson, the "add" button should appear, and if it _is_ currently in the lesson, the "remove" button should appear.

Reply:
I guess one thing I didn't discuss with @jtamiace is if this is an intermediary state, so if the "Add" is equivalent to selecting the resource, or if "Add" is equivalent to selecting _and saving_ the resource. I don't think it's clear in the spec. I'll ask in slack now and clarify.